src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -140,9 +140,14 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 	havingQual = masterQuery->havingQual;
 
 	/* estimate aggregate execution costs */
-	MemSet(&aggregateCosts, 0, sizeof(AggClauseCosts));
+	memset(&aggregateCosts, 0, sizeof(AggClauseCosts));
+#if (PG_VERSION_NUM >= 90600)
+	get_agg_clause_costs(NULL, (Node *) aggregateTargetList, AGGSPLIT_SIMPLE,
+						 &aggregateCosts);
+	get_agg_clause_costs(NULL, (Node *) havingQual, AGGSPLIT_SIMPLE, &aggregateCosts);
+#else
 	count_agg_clauses(NULL, (Node *) aggregateTargetList, &aggregateCosts);
-	count_agg_clauses(NULL, havingQual, &aggregateCosts);
```

Comment:
I'll give this a second look. I was leaning on the cost-calculating function to set pieces of `AggRef` correctly, but as it's getting pretty ugly, I might reimplement that portion of that method myself and call it in fewer places.